### DISCUSSION TITLE
GridLayout How to make equal row dimensions

### DISCUSSION THREAD

**Discussion by Firemanpl** [slint-ui/slint#3486] [General]
If i pushing Image .svg into Rectangle. Rows are adopting  to Images. I want that Images will adopt to Rows not the other way around.
```
export component Loading inherits Window {
icon: @image-url("src/assets/svg/xyz_logo.svg");
preferred-height: 100%;
preferred-width: 100%;
always-on-top: true;
background: #000;
GridLayout {
    Row{
        Rectangle {
            colspan: 3;
            border-color: yellow;
            border-width: 1px;
            Image {
                source: @image-url("src/assets/svg/zvf_logo.svg");
            }
        }
    }
    Row{
        Rectangle {
            colspan:3;
                Image {
                    image-fit: contain;
                    source: @image-url("src/assets/svg/sicuid_logo.svg");
                }

        }
    }

    Row{
        Rectangle {
            colspan: 3;
            border-color: cyan;
            border-width: 1px;

        }

    }
}
}

```

**Comment by hunger** [comment]
The only thing reporting a size in your code above is the image data. The Image elements defaults to that size when nothing else is set. The rectange surrounding the Image adapts to the size of the image, the Row to the size of the Rectangle, the GridLayout to the Rows and the Window to the Gridlayout.

You can e.g. set a `height` and `width` on the images themselves or the rectangles around them to change that.

**Comment by Firemanpl** [comment]
Ok, but How can I achieve the same Rows dimensions ? Because I want to for eg. set image to middle, without adopting Rows to rectangle and Rectangle to Image. How to make to adopt width and height to changing dimensions of whole window. If i resize window that all of dimensions elements will adopting to Window including all elements / object under Window / Page. (I don't know how to name this). I want to make responsive ui which can to match to every screen by fractal scaling interface and changing location elements of ui interface.

**Reply by hunger** [reply]
Check out [`Window::scale_factor`](https://slint.dev/releases/1.2.0/docs/rust/slint/struct.Window#method.scale_factor) for the fractual scaling. There is no need to change anything in the slint files (if you do not use physical units of course).

I am not sure I completely got what you want to do. Is [this](https://slintpad.com/?snippet=export+component+Loading+inherits+Window+%7B%0A%0A++++GridLayout+%7B%0A++++++++Row%7B%0A++++++++++++Rectangle+%7B%0A++++++++++++++++colspan%3A+3%3B%0A++++++++++++++++border-color%3A+yellow%3B%0A++++++++++++++++border-width%3A+1px%3B%0A++++++++++++++++Image+%7B%0A++++++++++++++++++++height%3A+50px%3B%0A++++++++++++++++++++source%3A+%40image-url%28%22https%3A%2F%2Fslint.dev%2Flogo%2Fslint-logo-full-light.svg%22%29%3B%0A++++++++++++++++%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++++++Row%7B%0A++++++++++++Rectangle+%7B%0A++++++++++++++++height%3A+50px%3B%0A++++++++++++++++colspan%3A3%3B%0A++++++++++++++++Image+%7B%0A++++++++++++++++++++height%3A+50px%3B%0A++++++++++++++++++++source%3A+%40image-url%28%22https%3A%2F%2Fupload.wikimedia.org%2Fwikipedia%2Fcommons%2Fd%2Fd5%2FRust_programming_language_black_logo.svg%22%29%3B%0A++++++++++++++++%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D) close?

**Comment by Firemanpl** [comment]
Ok. I solved this. 
`export component Loading inherits Window {
title: "dfd";
icon: @image-url("src/assets/svg/dfdf_logo.svg");
preferred-height: 100%;
preferred-width: 100%;
always-on-top: true;
background: #000;
Rectangle {
    drop-shadow-blur: 0px;
    drop-shadow-color:white;
Rectangle {
x: 0;
y: 0;
width: parent.width ;
height: parent.height / 3;

Image {
    source: @image-url("src/assets/svg/dfd_logo.svg");
    height: 5%;
}
}
Rectangle {
x: 0;
y: (parent.height - self.height) / 2;
width: parent.width;
height: parent.height / 3;


Image {
    source: @image-url("src/assets/svg/dfdf_logo.svg");
    height: 40%;
}
}
Rectangle {
x: 0;
y: (parent.height - self.height) / 1;
width: parent.width;
height: parent.height / 3;
}
}

}`

**Comment by ogoffart** [comment]
You can also use `%`  width and height in a layout to get that effect.

So `height:33.33%;` on every element should have the desired effect
